src/interfaces/python/swig_typemaps.i

```diff
@@ -1292,5 +1292,30 @@ TYPEMAP_SPARSEFEATURES_OUT(PyObject,      NPY_OBJECT)
     PyErr_SetString(PyExc_SystemError, $1.what());
     SWIG_fail;
 }
+%rename(_kernel) kernel;
+
+%pythoncode %{
+def _internal_factory_wrapper(object_name, new_name, docstring=None):
+    """
+    A wrapper that returns a generic factory that
+    accepts kwargs and passes them to shogun.object_name
+    via .put
+    """
+    _obj = getattr(_shogun, object_name)
+    def _internal_factory(name, **kwargs):
+
+        new_obj = _obj(name)
+        for k,v in kwargs.items():
+            new_obj.put(k, v)
+        return new_obj
+    if docstring:
+        _internal_factory.__doc__ = docstring
+    else:
+        _internal_factory.__doc__ = _obj.__doc__.replace(object_name, new_name)
```

Comment:
ah great catch.
So how does this work? sg.kernel docstring now is taken from the actualy "kernel" method? If so that is super nice and was something I was a bit worried about

Reply:
Yup!
```
In [1]: import shogun as sg                                                                                                                                                                                 

In [2]: kernel = sg.kernel("GaussianKernel", log_width=1.0)                                                                                                                                                 

In [3]: kernel                                                                                                                                                                                              
Out[3]: GaussianKernel(cache_size=10,combined_kernel_weight=1,lhs=null,lhs_equals_rhs=false,log_width=1,m_distance=EuclideanDistance(...),m_precomputed_distance=null,normalizer=IdentityKernelNormalizer(...),num_lhs=0,num_rhs=0,opt_type=0,optimization_initialized=false,properties=,rhs=null)

In [4]: kernel?                                                                                                                                                                                             
Type:        Kernel
String form: GaussianKernel(cache_size=10,combined_kernel_weight=1,lhs=null,lhs_equals_rhs=false,log_width=1,m <...> rmalizer(...),num_lhs=0,num_rhs=0,opt_type=0,optimization_initialized=false,properties=,rhs=null)
File:        ~/anaconda3/envs/Python36/lib/python3.6/site-packages/shogun.py
Docstring:  
The Kernel base class.

Non-mathematically spoken, a kernel is a function that given two input
objects ${\\bf x}$ and ${\\bf x'}$ returns a score describing the
similarity of the vectors. The score should be larger when the objects
are more similar.

It can be defined as

\\[ k({\\bf x},{\\bf x'})= \\Phi_k({\\bf x})\\cdot
\\Phi_k({\\bf x'}) \\]

where $\\Phi$ maps the objects into some potentially high
dimensional feature space.

Apart from the input features, the base kernel takes only one argument
(the size of the kernel cache) that is used to efficiently train
kernel-machines like e.g. SVMs.

In case you would like to define your own kernel, you only have to
define a new compute() function (and the kernel name via get_name()
and the kernel type get_kernel_type()). A good example to look at is
the GaussianKernel.

C++ includes: Kernel.h
```

Reply:
But it doesn't make a lot of sense as the signature has changed, and it should probably be `kernel(name, **kwargs) -> Kernel`. That's why added the option to pass a docstring manually

Reply:
Could we also move the docstring from `sg.kernel` to be the one of the c++ method `kernel` in factory? Or is that hard?

Reply:
Do you mean the one in Kernel.h?
```
/** @brief The Kernel base class.
 *
 * Non-mathematically spoken, a kernel is a function
...
```